recipes/mpdecimal/2.5.x/conanfile.py

```diff
@@ -0,0 +1,188 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+
+
+class MpdecimalConan(ConanFile):
+    name = "mpdecimal"
+    description = "mpdecimal is a package for correctly-rounded arbitrary precision decimal floating point arithmetic."
+    license = "BSD-2-Clause"
+    topics = ("conan", "mpdecimal", "multiprecision", "library")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://www.bytereef.org/mpdecimal"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "cxx": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "cxx": True,
+    }
+    exports_sources = "patches/**"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.arch not in ("x86", "x86_64"):
+            raise ConanInvalidConfiguration("Arch is unsupported")
+        if self.options.cxx:
+            if self.options.shared and self.settings.os == "Windows":
+                raise ConanInvalidConfiguration("A shared libmpdec++ is not possible on Windows (due to non-exportable thread local storage)")
+        else:
+            del self.settings.compiler.libcxx
+            del self.settings.compiler.cppstd
+
+    def build_requirements(self):
+        self.build_requires("autoconf/2.69")
```

Comment:
Not required for Visual Studio, or I missed something?
What about MinGW?

Reply:
Indeed, not needed for msvc.
mingw needs msys2. I didn't add it at first because I thought cci doesn't officially support mingw, but I have done it now so it's easier to build.